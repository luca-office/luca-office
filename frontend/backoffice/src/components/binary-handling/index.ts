export * from "./binary-form-placeholder/binary-form-placeholder"
export * from "./binary-update-modal/binary-entry"
export * from "./binary-update-modal/binary-update-modal"
export * from "./detail-view-binary/detail-view-binary"
export * from "./image-upload-modal/image-upload-modal"
