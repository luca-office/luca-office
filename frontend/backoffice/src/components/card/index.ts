// special backoffice card elements
export * from "./creation-card"
export * from "./duration-info/card-duration-info"
export * from "./tabbed-card/__mocks__/card-tabs.mock"
export * from "./tabbed-card/tabbed-card"
