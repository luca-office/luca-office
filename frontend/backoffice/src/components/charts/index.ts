export * from "./donut-chart/donut-chart-with-legend"
