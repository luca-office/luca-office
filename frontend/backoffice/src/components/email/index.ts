export * from "./email-body-meta-entry/email-body-meta-entry"
export * from "./email-body-setting-selection/email-body-setting-selection"
export * from "./email-content/email-content"
export * from "./email-directories-list/email-directories-list"
export * from "./email-header/email-header"
export * from "./email-unread-indicator/email-unread-indicator"
