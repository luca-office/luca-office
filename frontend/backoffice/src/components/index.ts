export * from "./binary-handling"
export * from "./card"
export * from "./card-overview/card-overview"
export * from "./charts"
export * from "./creation-bar/creation-bar"
export * from "./creation-status/creation-status"
export * from "./editable-multiline-textarea/editable-multiline-textarea"
export * from "./editing-status-indicator/editing-status-indicator"
export * from "./email"
export * from "./information-entry/information-entry"
export * from "./inline-editable-header/inline-editable-header"
export * from "./inline-editable-textarea/inline-editable-textarea"
export * from "./markdown-text/markdown-text"
export * from "./markdown-text-field/markdown-text-field"
export * from "./markdown-text-update-modal/markdown-text-update-modal"
export * from "./meta-entry/meta-entry"
export * from "./meta-entry-tags/meta-entry-tags"
export * from "./overlay-edit-field/overlay-edit-field"
export * from "./rating-progress-card/rating-progress-card"
export * from "./rating-progress-card/utils"
export * from "./resort-modal/resort-modal"
export * from "./sub-header-detail/sub-header-detail"
export * from "./sub-header-detail/sub-header-detail-container"
export * from "./sub-header-filter/sub-header-filter"
export * from "./table-action-bar/table-action-bar"
export * from "./upload-file-modal/upload-file-modal"
