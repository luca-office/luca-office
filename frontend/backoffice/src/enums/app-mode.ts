export enum AppMode {
  EDITOR = "editor",
  MANAGER = "manager",
  USERMANAGEMENT = "user_management",
  RATING = "rating"
}
