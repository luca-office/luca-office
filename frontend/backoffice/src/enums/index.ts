export * from "./app-mode"
export * from "./creation-status"
export * from "./email-status"
export * from "./email-update-key"
export * from "./entity-filter"
export * from "./questionnaire-detail-mode"
export * from "./scenario-update-keys"
export * from "./survey-timing"
export * from "./user-management"
