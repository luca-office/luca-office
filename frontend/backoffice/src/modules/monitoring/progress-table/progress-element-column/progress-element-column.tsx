import * as React from "react"
import {Icon, Text} from "shared/components"
import {IconName} from "shared/enums"
import {ProjectModuleProgressType} from "shared/graphql/generated/globalTypes"
import {ParticipantProjectProgress} from "shared/models"
import {Flex, TextSize} from "shared/styles"
import {useLucaTranslation} from "shared/translations"
import {getProjectModuleIcon} from "shared/utils"

export interface ProgressTableColumnProps {
  readonly progressElement: ParticipantProjectProgress
  readonly moduleCount: number
}

export const ProgressElementColumn: React.FunctionComponent<ProgressTableColumnProps> = ({
  progressElement,
  moduleCount
}) => {
  const {t} = useLucaTranslation()
  const currentElement = progressElement.moduleProgress.find(
    progress => progress.status === ProjectModuleProgressType.InProgress
  )
  const completedCount = progressElement.moduleProgress.filter(
    progress => progress.status === ProjectModuleProgressType.Completed
  ).length
  const isCompleted = completedCount === moduleCount
  const hasNotStarted =
    progressElement.moduleProgress.filter(progress => progress.status === undefined).length === moduleCount

  const icon = hasNotStarted
    ? IconName.Sandglass
    : isCompleted
    ? IconName.Check
    : currentElement
    ? getProjectModuleIcon(currentElement?.moduleType)
    : undefined

  return (
    <div css={Flex.row}>
      {icon && <Icon name={icon} hasSpacing={true} />}
      <Text size={TextSize.Medium}>
        {isCompleted
          ? t("dashboard__project_table_progress_done")
          : hasNotStarted
          ? t("dashboard__project_table_progress_none")
          : currentElement?.name}
      </Text>
    </div>
  )
}
