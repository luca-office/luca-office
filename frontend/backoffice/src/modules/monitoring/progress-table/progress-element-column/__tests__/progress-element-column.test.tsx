import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Icon, Text} from "shared/components"
import {IconName} from "shared/enums"
import {
  participantProjectProgressMock,
  participantProjectProgressStartedMock
} from "sharedTests/__mocks__/participant-project-progress.mock"
import {ProgressElementColumn, ProgressTableColumnProps} from "../progress-element-column"

const getComponent = (props?: Partial<ProgressTableColumnProps>) => (
  <ProgressElementColumn
    moduleCount={participantProjectProgressMock.moduleProgress.length}
    progressElement={participantProjectProgressMock}
    {...props}
  />
)

describe("progress-element-column", () => {
  it("renders correctly", () => {
    const component = getComponent()
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly (incomplete)", () => {
    const component = getComponent({progressElement: participantProjectProgressStartedMock})
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("has correct structure", () => {
    const component = shallow(getComponent())

    expect(component.find(Text)).toHaveLength(1)
    expect(component.find(Icon)).toHaveLength(1)
    expect(component.find(Icon).prop("name")).toEqual(IconName.Check)
    expect(component.find(Text).html()).toContain("dashboard__project_table_progress_done")
  })
  it("has correct structure (started)", () => {
    const component = shallow(
      getComponent({
        progressElement: participantProjectProgressStartedMock
      })
    )

    expect(component.find(Text)).toHaveLength(1)
    expect(component.find(Icon)).toHaveLength(1)
    expect(component.find(Icon).prop("name")).toEqual(IconName.Monitor)
    expect(component.find(Text).html()).toContain(participantProjectProgressStartedMock.moduleProgress[0].name)
  })
  it("has correct structure (not started)", () => {
    const component = shallow(
      getComponent({
        progressElement: {
          ...participantProjectProgressStartedMock,
          moduleProgress: [{...participantProjectProgressMock.moduleProgress[0], status: undefined}]
        },
        moduleCount: 1
      })
    )

    expect(component.find(Text)).toHaveLength(1)
    expect(component.find(Icon)).toHaveLength(1)
    expect(component.find(Icon).prop("name")).toEqual(IconName.Sandglass)
    expect(component.find(Text).html()).toContain("dashboard__project_table_progress_none")
  })
})
