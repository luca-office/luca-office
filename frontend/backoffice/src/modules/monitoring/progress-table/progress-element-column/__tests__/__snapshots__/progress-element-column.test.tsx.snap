// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress-element-column renders correctly (incomplete) 1`] = `
<div
  className="css-4fctw3-ProgressElementColumn"
>
  <div
    className="css-cj5xss-Icon"
    dangerouslySetInnerHTML={
      {
        "__html": {},
      }
    }
  />
  <div
    className="css-17zbm7y-Text"
  >
    Test Scenario
  </div>
</div>
`;

exports[`progress-element-column renders correctly 1`] = `
<div
  className="css-4fctw3-ProgressElementColumn"
>
  <div
    className="css-cj5xss-Icon"
    dangerouslySetInnerHTML={
      {
        "__html": {},
      }
    }
  />
  <div
    className="css-17zbm7y-Text"
  >
    dashboard__project_table_progress_done
  </div>
</div>
`;
